web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte

```diff
@@ -88,7 +87,22 @@
 
 {#if assets.length > 0}
   <div class="flex w-full flex-wrap gap-1 pb-20" bind:clientWidth={viewWidth}>
-    {#each assets as asset (asset.id)}
+    {#each assets.slice(0, -1) as asset (asset.id)}
+      <div animate:flip={{ duration: 500 }}>
+        <Thumbnail
+          {asset}
+          {thumbnailSize}
+          readonly={disableAssetSelect}
+          format={assets.length < 7 ? ThumbnailFormat.Jpeg : ThumbnailFormat.Webp}
+          on:click={(e) => (isMultiSelectionMode ? selectAssetHandler(e) : viewAssetHandler(e))}
+          on:select={selectAssetHandler}
+          selected={selectedAssets.has(asset)}
+          {showArchiveIcon}
+        />
+      </div>
+    {/each}
+
+    {#each assets.slice(-1) as asset (asset.id)}
```

Comment:
Why do we need a for each here? This will always be just one asset, right?

Reply:
Hmmm, good question. I probably just adapted the code above it. I'll change this to an `#if` block and check for a non-empty array for clarity.

Reply:
Oh, now I remember why I did this. It's because `animate:flip` seems to require `each`: `An element that uses the animate directive must be the immediate child of a keyed each block`. Should I remove that flag?

Reply:
I have no idea what that does lol. So up to you or someone else haha

Reply:
I don't know either so I think I'll just leave it alone for now lol

Reply:
I realized I can actually get the index of an asset in the each block and just conditionally fire the intersect event based on that.